RIAssigner/data/MatchMSData.py

```diff
@@ -35,8 +35,12 @@ def retention_indices(self) -> Iterable[Optional[int]]:
         raise NotImplementedError()
 
     @retention_indices.setter
-    def retention_indices(self, value: Iterable[int]):
-        raise NotImplementedError()
+    def retention_indices(self, values: Iterable[int]):
+        if len(values) == len(self._spectra):
+            for value_idx in range(len(values)):
+                self._spectra[value_idx].set(key="retentionindex", value=values[value_idx])

```

Comment:
Mapping has to be put inside *list* function, otherwise it doesn't evaluate what's inside. Nonetheless, I'd say it still looks better than the loop (https://github.com/RECETOX/RIAssigner/pull/7/commits/b16eed1350681dfbd41010f3778fccbf4a8952f3).